### DISCUSSION TITLE
host configuration set with ->set() not defined in task()->local()

### DISCUSSION THREAD

**Discussion by leevigraham** [deployphp/deployer#2229] [Help needed]
### Describe the bug

Given the following host / task:

```
host('localhost')
    ->set('stage', 'dev');

task('setup', function () {
   writeln(get('stage'));
})->local();
```

When I run `vendor/bin/dep setup dev` the output is:

```
 Configuration parameter `stage` does not exist.
```

I expected that any configuration value set on the host will be available.

I think this may be similar to #2133 

### Environment
- Deployer version: 6.8.0
- PHP version: 7.4
- Deployment target(s) OS: OSX Cantalina (Local dev machine)

**Comment by antonmedv** [comment]
No stage on local() tasks as stage config defined on other host.

**Comment by leevigraham** [comment]
I think what you're saying is that running a local task ignores host passed as a parameter in the `vendor/bin/dep setup dev`?

**Comment by antonmedv** [comment]
```php
host('localhost1');
host('localhost2');

task('setup', function () {
   // What host?
})->local();
```

I'm saying what there is no host selected at this task as it runs on local. And stage config is related to a host ¯\_(ツ)_/¯ 
Because of this, I refactored everything in v7, but is still need to know what are you running and there (host?).

**Comment by leevigraham** [comment]
To further confirm…

`vendor/bin/dep someTask` command still requires a host / stage argument even if you're running a single local task because there's no way to know if any of the tasks that will be called require a host.

**Comment by leevigraham** [comment]
I was expecting that even tho the `setup` task was flagged as local given I passed a host / stage argument the values set on that host / stage would be available in the task.

**Reply by antonmedv** [reply]
Sfage is confusing concept. So it already deleted in v7. If you selected one host. You may assume what local tasks also will get it config. One solution is to define global config what will return correct config for local task.
